sentry_sdk/integrations/gcp.py

```diff
@@ -0,0 +1,168 @@
+from datetime import datetime, timedelta
+from os import environ
+import sys
+
+from sentry_sdk.hub import Hub
+from sentry_sdk._compat import reraise
+from sentry_sdk.utils import (
+    capture_internal_exceptions,
+    event_from_exception,
+    logger,
+    TimeoutThread,
+)
+from sentry_sdk.integrations import Integration
+
+from sentry_sdk._types import MYPY
+
+# Constants
+TIMEOUT_WARNING_BUFFER = 1.5  # Buffer time required to send timeout warning to Sentry
+MILLIS_TO_SECONDS = 1000.0
+
+if MYPY:
+    from typing import Any
+    from typing import TypeVar
+    from typing import Callable
+    from typing import Optional
+
+    from sentry_sdk._types import EventProcessor, Event, Hint
+
+    F = TypeVar("F", bound=Callable[..., Any])
+
+
+def _wrap_func(func):
+    # type: (F) -> F
+    def sentry_func(*args, **kwargs):
+        # type: (*Any, **Any) -> Any
+
+        hub = Hub.current
+        integration = hub.get_integration(GcpIntegration)
+        if integration is None:
+            return func(*args, **kwargs)
+
+        # If an integration is there, a client has to be there.
+        client = hub.client  # type: Any
+
+        configured_time = int(environ.get("FUNCTION_TIMEOUT_SEC"))
```

Comment:
This is [failing in `mypy`](https://travis-ci.com/github/getsentry/sentry-python/jobs/375319774) because `environ.get` may return `None` if the env var is not set.

```py
In [1]: import os

In [2]: os.environ.get("FUNCTION_TIMEOUT_SEC")

In [3]: int(os.environ.get("FUNCTION_TIMEOUT_SEC"))
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-d034f2d5d55a> in <module>
----> 1 int(os.environ.get("FUNCTION_TIMEOUT_SEC"))

TypeError: int() argument must be a string, a bytes-like object or a number, not 'NoneType'
```

We need to handle the case in which the variable is not set.

If we don't know the configured timeout, we cannot start the `TimeoutThread`.

I suggest handling this by writing a debug log and then ignoring the rest of the wrapper, similar to when the integration is not in the current hub, as above.

Reply:
Okay, sure. I'll make this change.

Reply:
@rhcarvalho modified this.

Reply:
@rhcarvalho Maybe bcoz f the last check-in the linting got failed. I'll use make format and make lint. Though I do follow this as standard practice.

I'll check for the import errors in test jobs.

Reply:
@rhcarvalho I'm a little doubtful on updating of tox.ini file. I've updated it as per the reference from aws_lambda. But, it still gave the import error.
Are my changes appropriate?
Also, I've tried running the commands you suggested, it took a lot of time to run but didn't complete.

Reply:
currently CI fails because you are not declaring the dependency on `requests`

Reply:
you are missing an `tests/integrations/aws_lambda/__init__.py` which would skip the tests if the dependency is not installed (in other jobs)

Reply:
@untitaker okay, so I need a similar file for gcp tests right?